dockstore-integration-testing/src/test/java/io/dockstore/client/cli/BitBucketBasicIT.java

```diff
@@ -0,0 +1,467 @@
+/*
+ *    Copyright 2022 OICR and UCSC
+ *     
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *     
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *     
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package io.dockstore.client.cli;
+
+import static io.swagger.client.model.DockstoreTool.ModeEnum.MANUAL_IMAGE_PATH;
+import static junit.framework.TestCase.fail;
+import static org.junit.Assert.assertTrue;
+
+import io.dockstore.common.BitBucketTest;
+import io.dockstore.common.CommonTestUtilities;
+import io.dockstore.common.Registry;
+import io.swagger.client.ApiClient;
+import io.swagger.client.ApiException;
+import io.swagger.client.api.ContainersApi;
+import io.swagger.client.api.UsersApi;
+import io.swagger.client.model.DockstoreTool;
+import io.swagger.client.model.Tag;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.ExpectedSystemExit;
+import org.junit.contrib.java.lang.system.SystemErrRule;
+import org.junit.contrib.java.lang.system.SystemOutRule;
+import org.junit.experimental.categories.Category;
+
+/**
+ * Basic confidential integration tests, focusing on publishing/unpublishing both automatic and manually added tools
+ * This is important as it tests the web service with real data instead of dummy data, using actual services like Github and Quay
+ *
+ * @author aduncan
+ */
+@Category(BitBucketTest.class)
+public class BitBucketBasicIT extends BaseIT {
+    @Rule
+    public final ExpectedSystemExit systemExit = ExpectedSystemExit.none();
+
+    @Rule
+    public final SystemOutRule systemOutRule = new SystemOutRule().enableLog().muteForSuccessfulTests();
+
+    @Rule
+    public final SystemErrRule systemErrRule = new SystemErrRule().enableLog().muteForSuccessfulTests();
+
+    @Before
+    @Override
+    public void resetDBBetweenTests() throws Exception {
+        CommonTestUtilities.cleanStatePrivate1(SUPPORT);
+    }
+
+    @SuppressWarnings("checkstyle:parameternumber")
+    private DockstoreTool manualRegisterAndPublish(ContainersApi containersApi, String namespace, String name, String toolName,
```

Comment:
I think this method is now a duplicate of BasicIT::manualRegisterAndPublish. If I'm right, I think you should move one of the methods to BaseIT, and get rid of the duplication.

Can you also see if there's other stuff you can do the same with?

Reply:
I believe I moved all duplicate methods to BaseIT